### PR TITLE
Holtcssredux

### PR DESCRIPTION
**Pull request by mattretzer**
I think this css is ready; it's for Holt large NF.  Includes fullpage image support, runheads work on Prince 9 and are simplified, + other cleanup.
Known issues / possible enhancements:
• runhead does not appear on TOC pages, since there are no <p> elements on those pages.
• I did not add automatic insertion of inline image for copyright page, this pending checkin with Holt to see if it is always present and worth automating.

### CODE REVIEW

File: pdfmaker/css/holt/pdf.css [hunk 1]
```diff
@@ -1,9 +1,1175 @@
...
+content: counter(page, lower-roman) " • "; }
+
+section[data-type="chapter"] p + div.runheadright:after, 
+section[data-type="appendix"] p + div.runheadright:after {
+content: " • " counter(page); }
+
+section[data-type="chapter"] p + div.runheadright + div.runheadleft:before {
+content: counter(page) " • " string(title); }
+
+section[data-type="appendix"] p + div.runheadright + div.runheadleft:before {
```
Comment (nelliemckesson) [inline pdfmaker/css/holt/pdf.css]: Is it correct that there should be no text in the appendix header here?

File: pdfmaker/css/holt/pdf.css [hunk 2]
```diff
@@ -1,9 +1,1175 @@
...
+section[data-type="dedication"],
+section.adcard {
+  page-break-after: always; }
+
+section[data-type="halftitlepage"],
+section[data-type="titlepage"],
+section[data-type="preface"] {
+  page-break-before: right; }
+
+section[data-type="dedication"],
```
Comment (nelliemckesson) [inline pdfmaker/css/holt/pdf.css]: this is a duplicate (see line 264)

File: pdfmaker/css/holt/pdf.css [hunk 3]
```diff
@@ -1,9 +1,1175 @@
...
+  text-transform: lowercase;
+  font-size: 10pt;
+  line-height: 13pt;
+  float: bottom;
+  text-align: center;
+  text-indent: 0; }
+
+
+section[data-type="copyright-page"] p:first-of-type {
+  /*background-image: url("../../images/holt/HoltLogo30K.png");*/
```
Comment (nelliemckesson) [inline pdfmaker/css/holt/pdf.css]: are these comments still necessary? will they come in handy in the future? If so, you can leave them.

File: pdfmaker/css/holt/pdf.css [hunk 4]
```diff
@@ -1,9 +1,1175 @@
...
+figure + p[class*="AboutAuthorText"],
+p.AboutAuthorTextNo-Indentatatx1,
+p.IllustrationSourceis + p[class*="AboutAuthorText"] {
+  text-indent: 0; }
+
+section h1.AboutAuthorTextHeadatah + figure + p, section.abouttheauthor h1 + figure + p {
+  margin-top: 22pt; }
+
+section[data-type="preface"] *[class*="FMSubheadfmsh"]:first-of-type {
+  margin-top: -130.65pt; }
```
Comment (nelliemckesson) [inline pdfmaker/css/holt/pdf.css]: this is a big number! What are you trying to override here?

File: pdfmaker/css/holt/pdf.css [hunk 5]
```diff
@@ -1,9 +1,1175 @@
...
+section[data-type="preface"] *[class*="FMSubheadfmsh"]:first-of-type {
+  margin-top: -130.65pt; } 
+
+section[data-type="chapter"] {
+  page: chapter;
+  page-break-before: always;
+  prince-page-group: start;
+  counter-reset: footnote; }
+
+section[data-type="chapter"] ~ section[data-type="chapter"], {
```
Comment (nelliemckesson) [inline pdfmaker/css/holt/pdf.css]: unnecessary comma

File: pdfmaker/css/holt/pdf.css [hunk 6]
```diff
@@ -1,9 +1,1175 @@
...
+  display: block;
+  padding-top: 3pt;
+  font-size: 20pt;
+  line-height: 15pt;
+  font-weight: normal;
+  text-align: center;
+  margin-bottom: 10pt;
+  text-indent: 0; }
+
+[class*="Nonprinting"] {
```
Comment (nelliemckesson) [inline pdfmaker/css/holt/pdf.css]: add a * at the beginning of this def: \*[class*="Nonprinting"]. (It's totally possible this is a typo in my code as well...)

File: pdfmaker/scripts/holt/pdf.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    var thisParent = allParas[q];
+    var section = allParas[q].parentNode;
+    var i = 0;
+    var sectionHead = " ";
+    while (section.tagName != "SECTION" && section.tagName != "NAV" && i < 10) {
+      var section = section.parentNode;
+      i++;
+    };
+
+    var sectionType = section.getAttribute('data-type')
```
Comment (nelliemckesson) [inline pdfmaker/scripts/holt/pdf.js]: should there be a closing ";" on this line?

File: pdfmaker/scripts/holt/pdf.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+      } else if (section.getElementsByClassName('ChapTitleNonprintingctnp')[0] != null) {
+        var sectionHead = section.getElementsByClassName('ChapTitleNonprintingctnp')[0].textContent;
+      };
+      if (sectionHead.length < 4 && sectionHead.match(/^[0-9]+$/) != null) {
+        sectionHead = "Chapter " + sectionHead;
+      };
+    };  
+
+    var runHeadLeft = document.createElement("div");
+    runHeadLeft.setAttribute("class", "runheadleft")
```
Comment (nelliemckesson) [inline pdfmaker/scripts/holt/pdf.js]: needs closing ";" ?

File: pdfmaker/scripts/holt/pdf.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    runHeadLeft.textContent=sectionHead;
+    var runHeadRight = document.createElement("div");
+    runHeadRight.setAttribute("class", "runheadright")
```
Comment (nelliemckesson) [inline pdfmaker/scripts/holt/pdf.js]: needs closing ";" ?

File: pdfmaker/css/holt/pdf.css (hunk 1 above)
Comment (mattretzer) [inline pdfmaker/css/holt/pdf.css]: This part is right (the rest of the content is supplied by the JS) but this made me realize I had to edit the JS again - I had to insert empty divs for the left page chapter runheads :)

File: pdfmaker/css/holt/pdf.css (hunk 3 above)
Comment (mattretzer) [inline pdfmaker/css/holt/pdf.css]: I want to source it from github, but the url wasn't accurate until this is merged in.  Will go ahead and delete the other two lines..  good catch!

File: pdfmaker/css/holt/pdf.css (hunk 5 above)
Comment (mattretzer) [inline pdfmaker/css/holt/pdf.css]: thx

File: pdfmaker/scripts/holt/pdf.js (hunk 8 above)
Comment (mattretzer) [inline pdfmaker/scripts/holt/pdf.js]: yup

File: pdfmaker/css/holt/pdf.css (hunk 2 above)
Comment (mattretzer) [inline pdfmaker/css/holt/pdf.css]: One is page-break before, one after, right?

**Comment by mattretzer** [comment]
Hi @nelliemckesson , when I check the actual 'files changed' it looks like I already made the edits as discussed above... so this is ready to merge!

**Pull request merged**